File: analysers/WordCount/analyzer/src/main/java/cz/cuni/mff/socneto/storage/model/LogMessage.java
package cz.cuni.mff.socneto.storage.model;

import com.fasterxml.jackson.databind.node.ObjectNode;
import lombok.Builder;
import lombok.Data;

import java.util.Date;

@Data
@Builder
public class LogMessage {

    private String componentId;
    private EventType eventType;
    private String eventName;
    private String message;
    private Date timestamp;
    private ObjectNode attributes;

    public enum EventType {
        FATAL, ERROR, WARN, INFO, METRIC;
    }
}

File: backend/Domain/Models/LogMessage.cs
using System;
using Newtonsoft.Json.Linq;

namespace Socneto.Domain.Models
{
    public class LogMessage
    {
        
        public string ComponentId { get; set; }
        
        public string EventType { get; set; } // FATAL, ERROR, WARN, INFO, METRIC
        
        public string EventName { get; set; }
        
        public string Message { get; set; }
        
        public DateTime Timestamp { get; set; }
        
        public  JObject Attributes { get; set; }
        
    }
}